## Entendimiento de los datos

Durante esta fase, se lleva a cabo un proceso de comprensión de los datos con los que se trabajará. El conjunto de datos consiste en registros de mensajes de texto, los cuales se clasifican como "spam" o "ham". Estos registros están compuestos por dos columnas: "Category" y "Message".

Enlace al conjunto de datos: https://www.kaggle.com/datasets/team-ai/spam-text-message-classification?resource=download

In [6]:
import numpy as np
import pandas as pd

from matplotlib import pyplot as plt

Carga de datos

In [2]:
df = pd.read_csv("datos/SPAM text message 20170820 - Data.csv")
df.head()

Category                                            Message
0      ham  Go until jurong point, crazy.. Available only ...
1      ham                      Ok lar... Joking wif u oni...
2     spam  Free entry in 2 a wkly comp to win FA Cup fina...
3      ham  U dun say so early hor... U c already then say...
4      ham  Nah I don't think he goes to usf, he lives aro...

Se obtienen las dimensiones del dataset

In [4]:
print("Registros: {}".format(df.shape[0]))
print("Columnas: {}".format(df.shape[1]))

Registros: 5572
Columnas: 2


Se obtiene un resumen del dataset

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Category  5572 non-null   object
 1   Message   5572 non-null   object
dtypes: object(2)
memory usage: 87.2+ KB


Al analizar el resultado anterior, se observa que hay 5572 registros no nulos, lo que indica que no hay valores nulos en ninguna de las dos columnas.

Se verifica la cantidad de valores únicos en las etiquetas.

In [8]:
df["Category"].value_counts()

Category
ham     4825
spam     747
Name: count, dtype: int64

El resultado anterior revela que hay significativamente más etiquetas "ham" que etiquetas "spam", lo que indica un desequilibrio en las etiquetas. No se realizará ningún procesamiento sobre las etiquetas en este momento. Se observará si las redes neuronales pueden manejar este desequilibrio de clases.